### DISCUSSION TITLE
How to test using azure pipelines before create PR.

### DISCUSSION THREAD

**Discussion by shkwt** [renovatebot/renovate#12336] [Request Help] [answered]
As mentioned here, I want to test a branch before create pull request by renovate.

> With prCreation set to not-pending, Renovate waits until all tests have finished running, and only then creates the PR. When you receive the PR notification, you can take action immediately, as you have the full test results.

But, I couldn’t  find code description.
Could you please teach me how to set that on azure pipelines or renovate?

These are my codes:point_down: azure-pipelines.yml and renovate.json.

```.yml
schedules:
  - cron: '0 6 * * *'
    displayName: 'Daily morning renovate'
    branches:
      include: [master]

trigger: none

pool:
  vmImage: ubuntu-latest

steps:
  - task: npmAuthenticate@0
    inputs:
      workingFile: .npmrc

  - bash: |
      git config --global user.email 'bot@renovateapp.com'
      git config --global user.name 'Renovate Bot'
      npx --userconfig .npmrc renovate
    env:
      TOKEN: $(System.AccessToken)
```

```.json
{
  "labels": ["renovate"],
  "extends": ["config:base"],
  "reviewers": ["team: foobar"],
  "assignAutomerge": true,
  "rollbackPrs": true,
  "prCreation": "not-pending",
  "packageRules": [
    {
      "matchUpdateTypes": ["patch", "pin", "digest", "rollback"],
      "automerge": true
    },
    {
      "groupName": "@types",
      "packagePatterns": ["^@types/"],
      "automerge": true,
      "major": {
        "automerge": false
      },
      "addLabels": ["@types"]
    }
  ]
}
```

**Comment by rarkins** [comment, marked as answer]
The Renovate config is fine. It sounds like you need to configure Azure pipelines to run on any commit, not just on PRs.

**Reply by shkwt** [reply]
You mean when renovate commit, I have to set trigger it on azure pipelines?
And, will renovate wait until it finish?

**Reply by rarkins** [reply]
Renovate will create a commit in a `renovate/*` branch but with no PR. The PR will only be created once all tests pass or at least one fails (i.e. we know the result). This means Azure needs to be triggered by any commit, or at least any commit to `renovate/*` branches

**Reply by shkwt** [reply]
OK. I understand.
Thank you for your quick reply.
